### PR TITLE
Show stderr output in :messages

### LINKED ISSUE #37: Standard error output from the server should be shown
The Lean server prints many fatal and not-so-fatal errors on stderr (like stack overflows, panics, assertion failures etc.).  These should be shown somehow.

### PR DESCRIPTION
**Pull request by gebner**
Fixes #37.

### CODE REVIEW

File: lua/lean/stderr.lua [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+-- Show stderr output in :messages
+-- TODO: add upstream neovim API
+function stderr.enable()
+  local old_error = log.error
+  log.error = function(...)
+    local argc = select('#', ...)
+    if argc == 0 then return true end -- always enable error messages
+    if argc == 4 and select(1, ...) == 'rpc' and select(3, ...) == 'stderr'
+        and string.match(select(2, ...), 'lean') then
+      print(select(4, ...))
```
Comment (Julian) [inline lua/lean/stderr.lua:13]: Perhaps `vim.api.nvim_err_writeln` instead of `print`?

Comment (gebner) [reply lua/lean/stderr.lua]: Ok, changed!

**Comment by Julian** [comment]
lgtm!

**Comment by gebner** [comment]
Mmh, we could put the last ten lines in the infoview and then open a new buffer (with the full stderr output) on enter.

**Pull request merged**